### PR TITLE
Bug fix: 1. App will not crash on startup if location services are disabled on the device.

### PR DESCRIPTION
**Pull request by BarAgam**
Bug fix:
1. App will not crash on startup if location services are disabled on the device.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/120" target="_blank">ReviewNinja

File: data/src/main/java/com/teamagam/gimelgimel/data/location/LocationFetcher.java [hunk 1]
```diff
@@ -133,6 +137,16 @@ public void requestLocationUpdates() {
         mLocationManager.addGpsStatusListener(mStoppedGpsStatusDelegator);
     }
 
+    @SuppressWarnings("MissingPermission")
+    private void requestLocationUpdates(String provider) {
```
Comment (teamagam1) [inline data/src/main/java/com/teamagam/gimelgimel/data/location/LocationFetcher.java]: this is a private method and should be after all the publics

**Review by teamagam1 — CHANGES_REQUESTED** [review]
no one uses *isProviderExistsAndEnabled* method anymore, you can delete it.

File: data/src/main/java/com/teamagam/gimelgimel/data/location/LocationFetcher.java [hunk 2]
```diff
@@ -133,6 +137,16 @@ public void requestLocationUpdates() {
...
+        try {
```
Comment (teamagam1) [inline data/src/main/java/com/teamagam/gimelgimel/data/location/LocationFetcher.java]: what happens if it cannot add any provider? does the user knows those sort of things?
what if the users enables those options after the application initiated?

**Pull request merged**